extensions/amp-app-banner/0.1/amp-app-banner.js

```diff
@@ -0,0 +1,161 @@
+ */
+
+import {Layout} from '../../../src/layout';
+import {setStyles} from '../../../src/style';
+import {user} from '../../../src/log';
+import {platform} from '../../../src/platform';
+import {viewerFor} from '../../../src/viewer';
+import {CSS} from '../../../build/amp-app-banner-0.1.css';
+import {urlReplacementsFor} from '../../../src/url-replacements';
+import {xhrFor} from '../../../src/xhr';
+
+const TAG = 'amp-app-banner';
+
+class AmpAppBanner extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    if (platform.isIos()) {
+      this.preconnect.url('https://itunes.apple.com', onLayout);
+    } else if (platform.isAndroid()) {
+      this.preconnect.url('https://play.google.com', onLayout);
+    }
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.CONTAINER;
+  }
+
+  /** @override */
+  buildCallback() {
+    if (!platform.isIos() && !platform.isAndroid()) {
+      setStyles(this.element, {
+        'display': 'none',
+      });
+      return;
+    }
+
+    /** @private @const {!Viewer} */
+    const viewer = viewerFor(this.getWin());
+
+    /** @private @const {!Xhr} */
+    this.xhr_ = xhrFor(this.getWin());
+
+    /** @const @private {!Element} */
+    this.installLink_ = this.element.querySelector('a[install-link]');
+    user.assert(this.installLink_, '<a install-link> is required inside %s: %s',
+        TAG, this.element);
+
+    /** @const @private {!Element} */
+    this.openLink_ = this.element.querySelector('a[open-link]');
+    user.assert(this.openLink_, '<a open-link> is required inside %s: %s',
+        TAG, this.element);
+
+    if (platform.isIos()) {
+      const meta = this.getWin().document.head.querySelector(
+          'meta[name=apple-itunes-app]');
+      user.assert(meta,
+          '<meta name=apple-itunes-app> in document header is required: %s',
+          this.element);
+
+      // If non-embedded document and on Safari iOS just use the native app
+      // banner Safari supports if the meta tag is already setup.
+      if (!viewer.isEmbedded() && platform.isSafari()) {
+        setStyles(this.element, {
+          'display': 'none',
+        });
+        return;
+      }
+      this.parseIosMetaContent_(meta.getAttribute('content'));
+    }
+
+    // TODO: Provide a way to dismiss the banner and persist it.
+  }
+
+  /** @override */
+  layoutCallback() {
+    if (!platform.isAndroid()) {
+      return Promise.resolve();
+    }
+
+    const manifestLink = this.getWin().document.head.querySelector(
+        'link[rel=manifest],link[rel=amp-manifest]');
+    user.assert(manifestLink,
+        '<link rel=manifest> in the document head is required: %s',
+        this.element);
+
+    return this.xhr_.fetchJson(manifestLink.getAttribute('href'))
+        .then(response => {
+          this.parseManifest_(response);
+        }).catch(unusedError => {
+          // TODO: What do we do when we fail to fetch manifest.
```

Comment:
I chatted with @ericlindley-g about this. So I think publishers are already using the non-fixed position - medium for example. Event the native Safari iOS banner is non-fixed only the new Chrome on Android banner is fixed and its at the bottom of the page.